crates/dpki/src/seed.rs

```diff
@@ -278,6 +296,91 @@ impl DevicePinSeed {
     }
 }
 
+//--------------------------------------------------------------------------------------------------
+// Revocation Seed
+//--------------------------------------------------------------------------------------------------
+
+#[derive(Debug)]
+pub struct RevocationSeed {
+    inner: Seed,
+}
+
+impl SeedTrait for RevocationSeed {
+    fn seed(&self) -> &Seed {
+        &self.inner
+    }
+    fn seed_mut(&mut self) -> &mut Seed {
+        &mut self.inner
+    }
+}
+
+impl RevocationSeed {
+    /// Construct from a 32 bytes seed buffer
+    pub fn new(seed_buf: SecBuf) -> Self {
+        RevocationSeed {
+            inner: Seed::new_with_initializer(
+                SeedInitializer::Seed(seed_buf),
+                SeedType::Revocation,
+            ),
+        }
+    }
+
+    /// Generate a revocation key
+    pub fn generate_revocation_key(&mut self, derivation_index: u64) -> HcResult<KeyBundle> {
+        let mut ref_seed_buf = SecBuf::with_secure(SEED_SIZE);
+        let context = SeedContext::new(DEVICE_CTX);
+        let mut context = context.to_sec_buf();
+        kdf::derive(
+            &mut ref_seed_buf,
+            derivation_index,
+            &mut context,
+            &mut self.inner.buf,
+        )?;
+        Ok(KeyBundle::new_from_seed_buf(&mut ref_seed_buf)?)
+    }
+}
+
+//--------------------------------------------------------------------------------------------------
+// Auth Seed
+//--------------------------------------------------------------------------------------------------
+
+#[derive(Debug)]
+pub struct AuthSeed {
+    inner: Seed,
+}
+
+impl SeedTrait for AuthSeed {
+    fn seed(&self) -> &Seed {
+        &self.inner
+    }
+    fn seed_mut(&mut self) -> &mut Seed {
+        &mut self.inner
+    }
+}
+
+impl AuthSeed {
+    /// Construct from a 32 bytes seed buffer
+    pub fn new(seed_buf: SecBuf) -> Self {
+        AuthSeed {
+            inner: Seed::new_with_initializer(SeedInitializer::Seed(seed_buf), SeedType::Auth),
+        }
+    }
+
+    /// Generate a revocation key
+    pub fn generate_auth_key(&mut self, derivation_index: u64) -> HcResult<KeyBundle> {
+        let mut ref_seed_buf = SecBuf::with_secure(SEED_SIZE);
+        let context = SeedContext::new(DEVICE_CTX);
+        let mut context = context.to_sec_buf();
+        kdf::derive(
+            &mut ref_seed_buf,
+            derivation_index,
+            &mut context,
+            &mut self.inner.buf,
+        )?;
+        Ok(KeyBundle::new_from_seed_buf(&mut ref_seed_buf)?)
+    }
```

Comment:
You could probably generalize these two structs with an enum.

enum Seed
{ Auth,Revocation}